### ISSUE TITLE
Expose toolbelt::FileDescriptor?

### ISSUE THREAD

**Issue by mikael-s-persson** [dallison/subspace#8]
I'm not 100% sure what you prefer as a solution to this. Here is the problem:
I'm trying to use subspace subscribers and publishers, collect their fds from `GetPollFd()` (which returns `struct ::pollfd`), I then use them with `coroutine::Wait` (which takes a vector of `struct ::pollfd`), so far so good. But then, I also need to send them over `UnixSocket::SendFds` (which takes a vector of `toolbelt::FileDescriptor`). You see the issue, I cannot convert a `struct ::pollfd` into a `toolbelt::FileDescriptor` since that would close the fd on destruction.

There are a few solutions:
 - Have subspace pub/sub give out their `toolbelt::FileDescriptor` from `GetPollFd` (or some differently named functions).
 - Have a "release" function in `toolbelt::FileDescriptor` similar to `std::unique_ptr::release` that gives up ownership.
 - Overload `UnixSocket::SendFds` to take `int` fds, rather than demanding `toolbelt::FileDescriptor`.

Of course, all three could be done too. I think the last one is probably the most annoying and inefficient to implement.

**Comment by dallison** [comment]
I like the first option best.  I implemented the same as you did except I added 'const' to the functions and added a test.  Hope you don't mind that I check my version in (which is almost identical to yours).  I implemented it before I noticed you had also done it.

**Comment by dallison** [comment]
Yeah, good idea.  I've done that and released a new version of toolbelt.  Thanks.
